### PR TITLE
FEAT: /search command

### LINKED ISSUE #740: [FEAT] - Command to generate query URLs
### What feature would you like to see?

We should add a `/search` command which accepts:

- A VCS param, set to choices between GitHub and GitLab
- A language param, which is string

The command should return a query string using the VCS' search syntax. For example, `/search vsc:GitHub lang:TypeScript` should return `https://github.com/search?q=topic%3Ahacktoberfest%20language%3Atypescript&type=repositories` - note the inclusion of the hacktoberfest topic. 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Vyvy-vi**
# Pull Request

## Description:

 
Adds a /search command for finding projects with hacktoberfest tag

## Related Issue:

Closes #740

### CODE REVIEW

**Comment by Vyvy-vi** [comment]
Should the language option be replaced with a choices section?
The GitLab explore page allows searching by language only by some numerical ids... not sure if there's a way to this without hardcoding...

To ensure that this list is up to date, we could add a cronjob that scrapes this list from their explore page?)

File: src/helpers/getGitlabLanguageId.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const languages: { [key: string]: number } = {
```
Comment (naomi-lgbt) [inline src/helpers/getGitlabLanguageId.ts:1]: This should be extracted to a separate config file and exported.

File: src/helpers/getGitlabLanguageId.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Returns the ID used by GitHub to denote a language on their explore page.
+ *
+ * @param {string} lang The name of the language.
+ * @returns {number | undefined} The ID for the language, undefined otherwise.
+ */
+export const getGitlabLanguageId = (lang: string): number | undefined => {
+  if (lang in languages) {
+    return languages[lang];
+  }
```
Comment (naomi-lgbt) [inline src/helpers/getGitlabLanguageId.ts:38]:
```suggestion
return languages[lang] || "";
```

File: package-lock.json
Comment (naomi-lgbt) [inline package-lock.json:1]: Our projects use `pnpm` - this should be removed.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
> Should the language option be replaced with a choices section?

This isn't necessary. With the suggestions I've given, we'll use an empty string fallback if the language doesn't map to a code, which [works](https://gitlab.com/explore/projects/topics/hacktoberfest?language=).

> To ensure that [this list](https://github.com/nhcarrigan/weLoveHacktoberfest/pull/744/files#diff-616f43472ca30e799adb80071fecb4daa7911a6221b03ff124b55fea6f0e72b2) is up to date perhaps we could add a cronjob that scrapes this list from their explore page?
Not necessary. We can update it manually if needed - I'm not down for scraping. :3

**Comment by Vyvy-vi** [comment]
> This isn't necessary. With the suggestions I've given, we'll use an empty string fallback if the language doesn't map to a code, which [works](https://gitlab.com/explore/projects/topics/hacktoberfest?language=).

That's true, but for cases like "Jupyter Notebook", you wouldn't get the right results if you just type "Jupyter". "Objective C" would also lead to errors...

**Review by naomi-lgbt — APPROVED** [review]
This should be fine for now. We can expand on the language config in the future, if needed.

**Comment by naomi-lgbt** [comment]
## Thank You

Thank you for your continued contribution to our project. We have reviewed your pull request and are happy to accept these changes.

Please continue to watch for issues labelled `help wanted`, as these will be additional opportunities to contribute.

You can also see all open issues [through our contributor tool](https://contribute.naomi.lgbt)

**Pull request merged**